Detections/MultipleDataSources/Unauthorized_user_access_across_AWS_and_Azure.yaml

```diff
@@ -0,0 +1,147 @@
+            AWSAlertTime,
+            ActivityType,
+            Severity,
+            AWSAlertId,
+            AWSAlertTitle,
+            AWSAlertDescription,
+            AWSAlertLink,
+            Arn,
+            AWSresourceType,
+            AWSNetworkEntity,
+            AWSAlertUserNameEntity,
+            InstanceType,
+            APICallName,
+            APICallCount      
+        );
+        // Define a variable 'Azure_sigin' to collect Azure portal Signing activity from SigninLogs Table
+        let Azure_sigin = materialize (SigninLogs
+            | where AppDisplayName == "Azure Portal"
+            | where isnotempty(OriginalRequestId)
+            | summarize 
+                totalAzureLoginEventId = dcount(OriginalRequestId), 
+                AzureFailedEventsCount = dcountif(OriginalRequestId, ResultType != 0), 
+                AzureSuccessfulEventsCount = dcountif(OriginalRequestId, ResultType == 0),
+                AzureSetOfFailedEvents = makeset(iff(ResultType != 0, OriginalRequestId, ""), 5), 
+                AzureSetOfSuccessfulEvents = makeset(iff(ResultType == 0, OriginalRequestId, ""), 5) 
+            by 
+                IPAddress, 
+                UserPrincipalName, 
+                bin(TimeGenerated, 1min), 
+                UserAgent,
+                ConditionalAccessStatus,
+                OperationName,
+                RiskDetail,
+                AuthenticationRequirement,
+                ClientAppUsed 
+            // Extracting the name and UPN suffix from UserPrincipalName
+            | extend
+                Name = tostring(split(UserPrincipalName, "@")[0]),
+                UPNSuffix = tostring(split(UserPrincipalName, "@")[1])
+        );
+        // Join 'AwsAlert' and 'Azure_sigin' on the AWS Network Entity and Azure IP Address
+        AwsAlert
+        | join kind=inner Azure_sigin on $left.AWSNetworkEntity == $right.IPAddress
+entityMappings:
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: IPAddress
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: Name
+      - identifier: UPNSuffix
+        columnName: UPNSuffix
+customDetails:
+  AWSAlertUserName: AWSAlertUserNameEntity
+  AWSArn: Arn
+  AWSresourceType: AWSresourceType
+  AWSInstanceType: InstanceType
+  AWSAPICallName: APICallName
+  AWSAPICallCount: APICallCount
+  AzureUserAgent: UserAgent
+  AzureUser: UserPrincipalName
+  AzureClientAppUsed: ClientAppUsed
+  AzConditionalAccess: ConditionalAccessStatus
+  AzureOperationName: OperationName
+  AzureRiskDetail: RiskDetail
+  AzAuthRequirement: AuthenticationRequirement
+  alertSeverity: Severity
+alertDetailsOverride:
+  alertDisplayNameFormat: "{{AWSAlertTitle}}"
+  alertDescriptionFormat: "AWS Alert Description: '{{AWSAlertDescription}}' \n\n AWS ALert Link : '{{AWSAlertLink}}' \n\n Find More Details :https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_finding-types-active.html"

```

Comment:
You want to combine this with details from the description field in this template as well as other fields in the data (such as key entities). The aim of this description field is to given the triaging analyst as much information as possible to understand what they are looking at.

Reply:
We have a limit of only 2 or 3 variables we can call here so we can not add another entity variable here, 2 are already called

Reply:
It's not about the number of variables it's about the value this description brings to a user. If you saw this description as it is in an alert, would it provide all the details about to understand what the detection is looking for and what it has found? In its current state it doesn't do that.

Reply:
Ahh.. got your point, I just updated the alertDescriptionFormat details. kindly check